src/java/org/apache/cassandra/schema/SchemaConstants.java

```diff
@@ -77,14 +78,32 @@ public final class SchemaConstants
      */
     public static final int NAME_LENGTH = 48;
 
+    /**
+     * Longest acceptable file name. Longer names lead to too long file name error.
+     */
+    public static final int FILENAME_LENGTH = 255;
+
+    /**
+     * Longest acceptable table name, so it can be used in a directory
+     * name constructed with a suffix of a table id and a separator.
+     */
+    public static final int TABLE_NAME_LENGTH = FILENAME_LENGTH - 32 - 1;
+
     // 59adb24e-f3cd-3e02-97f0-5b395827453f
     public static final UUID emptyVersion;
 
     public static final List<String> LEGACY_AUTH_TABLES = Arrays.asList("credentials", "users", "permissions");
 
-    public static boolean isValidName(String name)
+    /**
+     * Validates that a name contains only alphanummeric characters or underscore,
```

Comment:
typo: alphanummeric

Reply:
I am not sure what typo is here. Can you clarify? (there is/was a typo in this work in another file, KeyspaceMetadata)

Reply:
alphanummeric should be with a single m

Reply:
Thank you. I confirm that it's so :) BTW you can use Suggestions in GH to make your suggestions more clear. E.g.:
```suggestion
     * Validates that a name contains only alphanumeric characters or underscore,
```